### ISSUE TITLE
Difficulty running cactus-graphmap-join 

### ISSUE THREAD

**Issue by NMarthe** [ComparativeGenomicsToolkit/cactus#993]
Hello, 
I'm working with the cactus pangenome pipeline and I've had some issues with the last step, cactus-graphmap-join. I get the error message "no segments in the gfa file" when running vg gbwt to make vg indexes, during the vg to gfa conversion step. 
From what I understand, this gfa file, clip.merged.gfa, comes from an other gfa file that I found in workDir and in jobStore, but the one in jobStore only has one line. 
After looking into the log, I saw the command "head -1" on said gfa file, which might be the reason why it has no segments. 
I copied the gfa from workDir in jobStore to replace the empty one and relaunched cactus-graphmap-join with the "--restart" option, and it worked fine. 

I am curious about why I had this issue, and if there is a cleaner way to do it. Could you maybe explain me what I did wrong ? 

I'm using cactus version 2.5.0.

The command I used : 
```
cactus-graphmap-join /scratch/AsianRiceGenomeGraph/steps/chr10/js5 --vg /scratch/AsianRiceGenomeGraph/steps/chr10/step_minigraph_chr10.vg --hal /scratch/AsianRiceGenomeGraph/steps/chr10/step_minigraph_chr10.hal --outDir /scratch/AsianRiceGenomeGraph/steps/chr10/ --outName graphe_steps_chr10 --reference IRGSP-1.0 --workDir /scratch/AsianRiceGenomeGraph/steps/chr10/workDir --clean never --cleanWorkDir never
```

The log from make_vg_indexes :
```
        [2023-04-18T23:55:54+0200] [MainThread] [I] [toil.worker] ---TOIL WORKER OUTPUT LOG---
        [2023-04-18T23:55:54+0200] [MainThread] [I] [toil] Running Toil version 5.9.2-54bfe0b146b76ecc6221de384c255e1be89547c6 on host node5.alineos.net.
        [2023-04-18T23:55:54+0200] [MainThread] [I] [toil.worker] Working on job 'make_vg_indexes' kind-make_vg_indexes/instance-9gol0_a4 v1
        [2023-04-18T23:55:55+0200] [MainThread] [I] [toil.worker] Loaded body Job('make_vg_indexes' kind-make_vg_indexes/instance-9gol0_a4 v1) from description 'make_vg_indexes' kind-make_vg_indexes/instance-9gol0_a4 v1
        [2023-04-18T23:55:55+0200] [MainThread] [I] [cactus.shared.common] Running the command ['grep', '-v', '^W     _MINIGRAPH_', '/scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/step_minigraph_chr10.vg.gfa']
        [2023-04-18T23:55:55+0200] [MainThread] [I] [toil-rt] 2023-04-18 23:55:55.093733: Running the command: "grep -v ^W     _MINIGRAPH_ /scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/step_minigraph_chr10.vg.gfa"
        [2023-04-18T23:55:55+0200] [MainThread] [W] [toil.lib.humanize] Deprecated toil method.  Please use "toil.lib.conversions.bytes2human()" instead."
        [2023-04-18T23:55:55+0200] [MainThread] [I] [toil-rt] 2023-04-18 23:55:55.117524: Successfully ran: "grep -v '^W     _MINIGRAPH_' /scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/step_minigraph_chr10.vg.gfa" in 0.0219 seconds and 1.0 Mi memory
        [2023-04-18T23:55:55+0200] [MainThread] [I] [toil-rt] 2023-04-18 23:55:55.117853: Running the command: "vg gbwt -G /scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/clip.merged.gfa --gbz-format -g /scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/clip.merged.gbz"
        [2023-04-18T23:55:55+0200] [MainThread] [W] [toil.fileStores.abstractFileStore] Failed job accessed files:
        [2023-04-18T23:55:55+0200] [MainThread] [W] [toil.fileStores.abstractFileStore] Downloaded file 'files/for-job/kind-vg_to_gfa/instance-bqsbiyc0/file-597104ab71eb47498350cfb53f6896eb/step_minigraph_chr10.vg.gfa' to path '/scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/step_minigraph_chr10.vg.gfa'
        Traceback (most recent call last):
          File "/home/cactus/cactus_env/lib/python3.10/site-packages/toil/worker.py", line 403, in workerScript
            job._runner(jobGraph=None, jobStore=jobStore, fileStore=fileStore, defer=defer)
          File "/home/cactus/cactus_env/lib/python3.10/site-packages/toil/job.py", line 2743, in _runner
            returnValues = self._run(jobGraph=None, fileStore=fileStore)
          File "/home/cactus/cactus_env/lib/python3.10/site-packages/toil/job.py", line 2660, in _run
            return self.run(fileStore)
          File "/home/cactus/cactus_env/lib/python3.10/site-packages/toil/job.py", line 2888, in run
            rValue = userFunction(*((self,) + tuple(self._args)), **self._kwargs)
          File "/home/cactus/cactus_env/lib/python3.10/site-packages/cactus/refmap/cactus_graphmap_join.py", line 568, in make_vg_indexes
            cactus_call(parameters=['vg', 'gbwt', '-G', merge_gfa_path, '--gbz-format', '-g', gbz_path])
          File "/home/cactus/cactus_env/lib/python3.10/site-packages/cactus/shared/common.py", line 839, in cactus_call
            raise RuntimeError("{}Command {} exited {}: {}".format(sigill_msg, call, process.returncode, out))
        RuntimeError: Command /usr/bin/time -v vg gbwt -G /scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/clip.merged.gfa --gbz-format -g /scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/0cb2/5f1b/tmpj_lrfq3w/clip.merged.gbz exited 134: stdout=None, stderr=terminate called after throwing an instance of 'std::runtime_error'
          what():  No segments in the GFA file
        ERROR: Signal 6 occurred. VG has crashed. Visit https://github.com/vgteam/vg/issues/new/choose to report a bug.
        Stack trace path: /tmp/vg_crash_7AfSSr/stacktrace.txt
        Please include the stack trace file in your bug report!
        Command exited with non-zero status 134
```

The "head" command I mentionned if it helps :
```
[2023-04-18T23:53:30+0200] [MainThread] [I] [toil-rt] 2023-04-18 23:53:30.223259: Running the command: "head -1 /scratch/AsianRiceGenomeGraph/steps/chr10/workDir/72b2a22953dd579e83d64dd51304b118/dcae/6f0d/tmpw_76i2pp/step_minigraph_chr10.vg.gfa"
```

Let me know if you need the complete log, or anything else. 

Thank you for your help,

Nina

**Comment by glennhickey** [comment]
I've never seen this error.  If you are able to share `step_minigraph_chr10.vg ` with me, I will try to reproduce.  I suspect it's an issue with sample names which may be somehow causing your entire graph to be filtered out. 

Another idea is to try switching to the new `cactus-pangenome` interface, which makes it a little more difficult to, say, pass different values to `--reference` for different steps of the pipeline which could lead to weird errors like this.

**Comment by NMarthe** [comment]
Sure, here it is : https://drive.google.com/file/d/1jqQgAZC9ts9rIh0JbmeHelQ2x8LhajGI/view?usp=share_link
Let me know if you have trouble downloading it. 
I also had an error with an out of range ascii character, so it might be the sample names indeed. I solved it by reinstalling cactus in a singularity. 
I initially tried using cactus-pangenome, but I got the same error. I double-checked my script, and the --reference value is consistent in all the steps.
